autoload/rust.vim

```diff
@@ -472,7 +472,8 @@ function! rust#Test(all, options) abort
     let pwd = shellescape(pwd)
 
     if a:all
-        execute '!cd ' . pwd . ' && cargo test ' . a:options
+        execute 'silent !cd' pwd
+        execute '!cargo test' a:options
```

Comment:
I think this change does not work since `!cd` just execute `cd` command in sub shell and it does not affect Vim's current directory.

How about this?

```vim
let cd = haslocaldir() ? 'lcd' : 'cd'
let saved = getcwd()
try
    silent execute cd pwd
    execute '!cargo test' a:options
finally
    silent execute cd saved
endtry
```

This changes current working directory temporarily and restores it after.

Please also remove above `let pwd = shellescape(pwd)` line since it is not necessary for `:cd` or `:lcd` command.